microsetta_admin/metadata_util.py

```diff
@@ -1,141 +1,258 @@
-                    cached_templates[answer["template"]] = None
-                    continue
-
-            cached_response_json[sample_barcode].append(answer)
-
-    multiselect_map = _build_possible_values_per_question(cached_templates)
-    transformed_response = defaultdict(dict)
-
-    for barcode in cached_response_json:
-        transformed_response[barcode]["host_subject_id"] = \
-            barcode_to_host_sub_id[barcode]
-        for survey_answers in cached_response_json[barcode]:
-            for q_id in survey_answers['response']:
-                col_name = survey_answers["response"][q_id][0]
-                answer = survey_answers["response"][q_id][1]
-                if type(answer) is list:
-                    answer_set = {a: False for a in answer}
-                    for possible_answer in multiselect_map[q_id]:
-                        multi_col_name = _build_col_name(col_name,
-                                                         possible_answer)
-                        if multi_col_name in transformed_response[barcode]:
-                            error_report.append({
-                                "barcode": barcode,
-                                "error": "Duplicate response to question " +
-                                         str(q_id) + "/" + str(col_name)
-                            })
-                        else:
-                            transformed_response[barcode][multi_col_name] = \
-                                possible_answer in answer_set
-                            answer_set[possible_answer] = True
-                    for a in answer_set:
-                        if not answer_set[a]:
-                            error_report.append({
-                                "barcode": barcode,
-                                "error": "Invalid response to question " +
-                                         str(q_id) + "/" + str(col_name)
-                            })
-                else:
-                    if col_name in transformed_response[barcode]:
-                        error_report.append({
-                            "barcode": barcode,
-                            "error": "Duplicate response to question " +
-                                     str(q_id) + "/" + str(col_name)
-                        })
-                    else:
-                        transformed_response[barcode][col_name] = answer
-
-    # print("TEMPLATES")
-    # print(json.dumps(cached_templates, indent=2))
-    # print("RESPONSES")
-    # print(json.dumps(cached_response_json, indent=2))
-    # print("TEMPLATE POSSIBLE MULTISELECT OPTIONS")
-    # print(json.dumps(multiselect_map, indent=2))
-    # print("TRANSFORMED")
-    # print(json.dumps(transformed_response, indent=2))
-    # print("ERRORS")
-    # print(json.dumps(error_report, indent=2))
-
-    return transformed_response, error_report
-
-
-def _build_possible_values_per_question(templates):
```

Comment:
How do you determine what columns you need for multiselect questions without this function?

Reply:
Based on the data in the survey responses. So your earlier observation is correct that if in the set of barcodes, not all questions are responded to then some multi-selects will be omitted

Reply:
I'll pulldown a full version of the survey template to assess all multiselect questions

Reply:
@dhakim87 this is now resolved